File: task-03/Python/3/Sub-Task3.py
n = int(input("Enter a number: "))

for i in range(n-1):
    for j in range(i,n):
        print(" ",end="")
    for k in range(i):
        print("*",end="")
    for l in range(i+1):
        print("*",end="")
    print()
for a in range(n):
    for b in range(a+1):
        print(" ",end="")
    for c in range(a,n-1):
        print("*",end="")
    for d in range(a,n):
        print("*",end="")
    print()

File: task-03/Haskell/3/Sub-Task3.hs
import Control.Monad (forM_)

main :: IO ()
main = do
    putStrLn "Enter a number:"
    input <- getLine
    let n = read input :: Int
    printDiamond n

printDiamond :: Int -> IO ()
printDiamond n = do
    forM_ [0..(n-1)] $ \i -> do
        putStr (replicate (n - i - 1) ' ')
        putStr (replicate (2 * i + 1) '*')
        putStrLn ""
    
    forM_ [(n-2),(n-3)..0] $ \i -> do
        putStr (replicate (n - i - 1) ' ')
        putStr (replicate (2 * i + 1) '*')
        putStrLn ""
